src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -766,5 +767,45 @@ private async Task<bool> CompleteOutstandingActiveRequests(string targetName)
 
             return false;
         }
+
+        private void ComputeAfterTargetFailures(string[] targetNames)
+        {
+            // Cache intermediate results so we don't visit a target more than once.
+            Dictionary<string, bool> resultsWithAfterTargets = new Dictionary<string, bool>();
+
+            foreach (string targetName in targetNames)
```

Comment:
I'd personally vote for common sense rather than strict rules based on such things as loop depth and cyclomatic complexity. In this particular case I do see methods with 3 nested loops in the code base so I would assume it's ok.

Reply:
No existing rules.